File: src/com/tzavi/fileorganizer/config/RuleManager.java
package com.tzavi.fileorganizer.config;

import java.util.HashMap;
import java.util.Map;

import com.tzavi.fileorganizer.ui.MenuManager;

public class RuleManager {
    public static Map<String, String> rules = new HashMap<>();

    public static void setRules() {
        // compressed
        rules.put("7z", "compressed");
        rules.put("rar", "compressed");
        rules.put("zip", "compressed");
        // executables
        rules.put("exe", "executables");
        // images
        rules.put("jpg", "images");
        rules.put("jpeg", "images");
        rules.put("png", "images");
        rules.put("gif", "images");
        // videos
        rules.put("mp4", "videos");
        // documents
        rules.put("pdf", "documents");
        rules.put("docx", "documents");
        rules.put("txt", "documents");
    }

    public static void setNewRule() {
        MenuManager.cleanscreen();
        System.out.println("==================================================");
        System.out.println("          Add new extension and folder");
        System.out.println("For the extension dont type the '.' write only the name");
        System.out.println("             (e.g., exe or zip)");
        System.out.println("==================================================");
        System.out.println("Extension: ");
        System.out.println("==================================================");
        String extension = MenuManager.scanner.nextLine().trim().toLowerCase();
        System.out.println("==================================================");
        System.out.println("Folder name: ");
        System.out.println("==================================================");
        String folderName = MenuManager.scanner.nextLine();
        System.out.println("==================================================");
        System.out.printf("| Added: | Extension: '%s' | To | Folder: '%s' |\n", extension, folderName);
        System.out.println("==================================================");
        rules.put(extension, folderName);
        System.out.println("Press Enter to continue: ");
        System.out.println("==================================================");
        MenuManager.scanner.nextLine();
    }

    public static void listRules() {
        MenuManager.cleanscreen();
        System.out.println("==================================================");
        System.out.println("       | Listing supported extensions... |");
        System.out.println("==================================================");
        rules.forEach((extension, folder) -> {
            System.out.printf("| Extension: '%s' | -> | Folder: '%s' |\n", extension, folder);
            System.out.println("==================================================");
        });
    }
}

File: Config/RuleManager.cs
using FileOrganizer.FileOrganizer.Settings;
using FileOrganizer.FileOrganizer.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace FileOrganizer.FileOrganizer.Config
{
    internal class RuleManager
    {
        public static Dictionary<string, string> rules = new Dictionary<string, string>();

        public static void setRules()
        {
            // compressed
            rules.Add("7z", "compressed");
            rules.Add("rar", "compressed");
            rules.Add("zip", "compressed");
            // executables
            rules.Add("exe", "executables");
            // images
            rules.Add("jpg", "images");
            rules.Add("jpeg", "images");
            rules.Add("png", "images");
            rules.Add("gif", "images");
            // videos
            rules.Add("mp4", "videos");
            // documents
            rules.Add("pdf", "documents");
            rules.Add("docx", "documents");
            rules.Add("txt", "documents");
        }

        public static void setNewRule()
        {
            MenuManager.CleanScreen();
            Console.WriteLine("=====================================================");
            Console.WriteLine("           Add new extension and folder");
            Console.WriteLine("   For the extension dont type the '.' write only the name");
            Console.WriteLine("               (e.g., exe or zip)");
            Console.WriteLine("=====================================================");
            Console.WriteLine("Extension: ");
            Console.WriteLine("=====================================================");
            String extension = Console.ReadLine().Trim().ToLower();
            if (string.IsNullOrWhiteSpace(extension))
            {
                extension = "Non-Specified extension";
            }
            Console.WriteLine("=====================================================");
            Console.WriteLine("Folder Name: ");
            Console.WriteLine("=====================================================");
            String folderName = Console.ReadLine();
            if (string.IsNullOrWhiteSpace(folderName))
            {
                folderName = "default_folder";
            }
            Console.WriteLine("=====================================================");
            Console.WriteLine($"| Added: | Extension: '{extension}' | To | Folder '{folderName}' |");
            Console.WriteLine("=====================================================");
            rules[extension] = folderName;
            Console.WriteLine("Press ENTER to continue: ");
            Console.WriteLine("=====================================================");
            Console.ReadLine();
            AppSettings.SaveRules();
        }

        public static void listRules()
        {
            MenuManager.CleanScreen();
            Console.WriteLine("=====================================================");
            Console.WriteLine("            Listing supported extensions...");
            Console.WriteLine("=====================================================");
            foreach (var rule in rules)
            {
                Console.WriteLine($"| Extension: {rule.Key,-3} | -> | Folder: {rule.Value,-3} | ");
                Console.WriteLine("=====================================================");
            }
        }
    }
}